processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -533,6 +535,12 @@ public byte[] computeCacheKey(GroupByQuery query)
             .appendCacheable(query.getVirtualColumns());
         if (query.isApplyLimitPushDown()) {
           builder.appendCacheable(query.getLimitSpec());
+          if (query.getLimitSpec() instanceof DefaultLimitSpec) {
+            DefaultLimitSpec limitSpec = (DefaultLimitSpec) query.getLimitSpec();
+            for (OrderByColumnSpec column : limitSpec.getColumns()) {
+              appendOrderingColumnCacheable(query, builder, column.getDimension());
```

Comment:
Yeah that makes sense, I hesitated on doing an approach involving passing the `Query` in initially because these ordering/limiting containers currently implement `Cacheable` and wasn't quite sure what that would look like, but didn't consider also passing the builder in which is probably the best thing to do here :+1: